daemon/oci_linux.go

```diff
@@ -911,6 +926,31 @@ func (daemon *Daemon) createSpec(c *container.Container) (retSpec *specs.Spec, e
 		s.Linux.ReadonlyPaths = c.HostConfig.ReadonlyPaths
 	}
 
+	// We just need to passthrough two types of annotations to runtime
+	// first is user specific annotations which with prefix of `annotations.`
+	// the other is labels for docker used which is marked as internal.
+	for k, v := range c.Config.Labels {
+		logrus.Info(k)
+		if oldValue, ok := s.Annotations[k]; ok {
+			return nil, fmt.Errorf("Key %q already exisit in Annotations, the new value will be ignored,"+
+				"oldvalue is %q, new value is %q", k, oldValue, v)
+		}
+		// user specific annotations
+		if strings.HasPrefix(k, anntotaionPrefix) {
+			key := strings.TrimPrefix(k, anntotaionPrefix)
+			logrus.Info(key)
+			s.Annotations[key] = v
+		}
+
+		// interl labels for docker
+		for _, internalKey := range internalLabelKeys {
+			if k == internalKey {
+				s.Annotations[k] = v
+				break
+			}
+		}
+	}
+
 	return &s, nil
```

Comment:
seems the code base of this RP is out of date

Reply:
I check the mainline and I think it's up to date...  ^_^